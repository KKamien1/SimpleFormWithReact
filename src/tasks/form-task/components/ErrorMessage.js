import React from 'react'

export default function ErrorMessage({ errors: [error] = [] }) {
    return error ? (
        <span className='form__error-message'>
            {error}
        </span>
    ) : null;
}
