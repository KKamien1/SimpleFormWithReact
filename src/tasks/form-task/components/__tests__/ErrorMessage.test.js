import React from 'react';
import { shallow } from 'enzyme';

import ErrorMessage from '../ErrorMessage'

describe('ErrorMessage component', () => {
    test('should match to snapshot', () => {
        const component = shallow(<ErrorMessage errors={['Test message']} />);
        expect(component.html()).toMatchSnapshot();
    })
    test('should return null if there is no message', () => {
        const component = shallow(<ErrorMessage errors={[]} />);
        expect(component.html()).toBe(null);
    })

})
