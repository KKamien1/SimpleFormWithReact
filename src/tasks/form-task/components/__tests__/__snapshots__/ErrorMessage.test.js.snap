// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component should match to snapshot 1`] = `"<span class=\\"form__error-message\\">Test message</span>"`;
